src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotContainTypeNames.cs

```diff
@@ -84,34 +85,51 @@ public override void Initialize(AnalysisContext analysisContext)
         {
             analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
-
-            // Analyze named types and fields.
-            analysisContext.RegisterSymbolAction(symbolContext =>
+            
+            analysisContext.RegisterCompilationStartAction(compilationStartAnalysisContext =>
             {
-                AnalyzeSymbol(symbolContext.Symbol, symbolContext);
-            }, SymbolKind.NamedType, SymbolKind.Field);
+                INamedTypeSymbol equalityComparerType = WellKnownTypes.GenericIEqualityComparer(compilationStartAnalysisContext.Compilation);
+                IMethodSymbol iEqualityComparerGetHashCodeMethod = equalityComparerType?.GetMembers("GetHashCode").Single() as IMethodSymbol;
 
-            // Analyze properties and methods, and their parameters.
-            analysisContext.RegisterSymbolAction(symbolContext =>
-            {
-                if ((symbolContext.Symbol as IMethodSymbol).IsEqualsOverride())
+                // Analyze named types and fields.
+                compilationStartAnalysisContext.RegisterSymbolAction(symbolContext =>
                 {
-                    // Ignore, given the method signature and parameter names are auto-generated by the IDE:
-                    // public override bool Equals(object obj)
-                    return;
-                }
+                    AnalyzeSymbol(symbolContext.Symbol, symbolContext);
+                }, SymbolKind.NamedType, SymbolKind.Field);
 
-                AnalyzeSymbol(symbolContext.Symbol, symbolContext);
+                // Analyze properties and methods, and their parameters.
+                compilationStartAnalysisContext.RegisterSymbolAction(symbolContext =>
+                {
+                    var method = symbolContext.Symbol as IMethodSymbol;
+                    if (method != null)
+                    {
+                        if (method.IsEqualsOverride())
+                        {
+                            // Ignore, given the method signature and parameter names are auto-generated by the IDE:
+                            // public override bool Equals(object obj)
+                            return;
+                        }
 
-                ImmutableArray<IParameterSymbol> parameters = symbolContext.Symbol.Kind == SymbolKind.Property ?
-                    ((IPropertySymbol)symbolContext.Symbol).Parameters :
-                    ((IMethodSymbol)symbolContext.Symbol).Parameters;
+                        if (IsIEqualityComparerGetHashCodeInterfaceImplementation(method, iEqualityComparerGetHashCodeMethod))
```

Comment:
I believe not, because
```csharp
public override int GetHashCode() { }
```

is not what we are looking for here. Instead we are trying to find a signature like:

```csharp
public int GetHashCode(string obj) { }
```

or even:

```csharp
int IEqualityComparer<string>.GetHashCode(string obj)
```